### PR TITLE
Upgrade k/k dependency to v1.20

### LINKED ISSUE #1080: [occm] Investigate to remove vendored code from occm
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind feature

I've created this issue to discuss and investigate if we can switch to fully use the k/k `cloud-controller-manager` code as seen [here](https://github.com/kubernetes/kubernetes/blob/master/cmd/cloud-controller-manager/controller-manager.go), and not use our own implementation of parts of the k/k `cloud-controller-manager` in [this](https://github.com/kubernetes/cloud-provider-openstack/blob/master/cmd/openstack-cloud-controller-manager/main.go) part of the `openstack-cloud-controller-manager`. 

I've been reading up on and looking a lot on the different external cloud controller managers out there as well as the packages and source code that makes up the cloud-controller-manager and it's internals.

Today the `main()` entrypoint looks like [this](https://github.com/kubernetes/cloud-provider-openstack/blob/master/cmd/openstack-cloud-controller-manager/main.go). My concern is that we have vendored code within this repository that we actually should be used by consuming the `k8s.io/kubernetes/cmd/cloud-controller-manager/app`. If i compare the code of the init function of the different controllers within the cloud-controller-manager it has actually diverged already:

https://github.com/kubernetes/kubernetes/blob/356c121c6d16240b4f5ad7dc58c3065b9adca28e/cmd/cloud-controller-manager/app/core.go#L105

vs

https://github.com/kubernetes/cloud-provider-openstack/blob/3909137e68519733d9832ef2dabd812550ca6884/cmd/openstack-cloud-controller-manager/main.go#L227-L259

I'm _almost_ convinced that we don't need more than the [k/k cloud-controller-manager `main.go`](https://github.com/kubernetes/kubernetes/blob/master/cmd/cloud-controller-manager/controller-manager.go) code to run our (`openstack-cloud-controller-manager`). Of course there's parts that we'll need to add but the majority of the code can be removed. So basically there's no need for **us** to maintain that code!

Questions:
* Does this make any sense?
* Is there any good reasons for keeping that code vendored instead of consuming the upstream? 

**What happened**:
N/A

**What you expected to happen**:
N/A

**How to reproduce it**:
N/A

**Anything else we need to know?**:
N/A

**Environment**:
N/A

### PR DESCRIPTION
**Pull request by ramineni**
**What this PR does / why we need it**:
This PR updates k/k dependency to 1.20

**Which issue this PR fixes(if applicable)**:
fixes #1080

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/openstack-cloud-controller-manager/main.go [hunk 1]
```diff
@@ -101,21 +95,30 @@ the cloud specific control loops shipped with Kubernetes.`,
...
 
-			if err := app.Run(c.Complete(), wait.NeverStop); err != nil {
+			cloudconfig := c.Complete().ComponentConfig.KubeCloudShared.CloudProvider
+			cloud, err := cloudprovider.InitCloudProvider(cloudconfig.Name, cloudconfig.CloudConfigFile)
+			if err != nil {
+				fmt.Fprintf(os.Stderr, "%v\n", err)
+			}
+			if cloud == nil {
+				fmt.Fprintf(os.Stderr, "%v\n", err)
+			}
```
Comment (jichenjc) [inline cmd/openstack-cloud-controller-manager/main.go:111]: I assume we should error out here and line 107?

File: pkg/cloudprovider/.import-restrictions [hunk 2]
```diff
@@ -7,7 +7,8 @@
 				"k8s.io/cloud-provider",
 				"k8s.io/component-base",
 				"k8s.io/klog",
-				"k8s.io/utils"
+				"k8s.io/utils",
+                                "k8s.io/mount-utils"
```
Comment (jichenjc) [inline pkg/cloudprovider/.import-restrictions:11]: intent seems can be updated

File: cmd/openstack-cloud-controller-manager/main.go (hunk 1 above)
Comment (ramineni) [reply cmd/openstack-cloud-controller-manager/main.go:111]: right. Ill update. Thanks

**Comment by ramineni** [comment]
@jichenjc There is some issue , we have errors in ccm logs , unable to contact api server , still not exactly sure what is the issue

**Comment by jichenjc** [comment]
> @jichenjc There is some issue , we have errors in ccm logs , unable to contact api server , still not exactly sure what is the issue.
> strange happening only for keystone job. @lingxiankong could you have a look?

I think in general we need have more logs including 
devstack, all pods log etc ,will take a look at what I can do after holiday 
it should be something like 

https://logs.openlabtesting.org/logs/12/712/7456fdaa676ab2c2da8b71dc79d204d8519a0107/check/cluster-api-provider-openstack-current-acceptance-test-v1.18/3784683/logs/_artifacts/logs/

https://logs.openlabtesting.org/logs/12/712/7456fdaa676ab2c2da8b71dc79d204d8519a0107/check/cluster-api-provider-openstack-current-acceptance-test-v1.18/3784683/logs/

so we will have more info

**Comment by lingxiankong** [comment]
I'm looking at this issue

File: pkg/cloudprovider/.import-restrictions (hunk 2 above)
Comment (ramineni) [reply pkg/cloudprovider/.import-restrictions:11]: seems indent is correct when u open raw file

**Comment by ramineni** [comment]
@jichenjc could you review? I guess we can cut release branch , once this is merged

**Comment by jichenjc** [comment]
> @jichenjc could you review? I guess we can cut release branch , once this is merged

sorry, just saw this : ) 
glad it's merged already

**Pull request merged**